common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/TaskanaConfigurationInitializer.java

```diff
@@ -158,6 +163,89 @@ Optional<T> initialize(
         Properties properties, String separator, Field field, TaskanaProperty taskanaProperty);
   }
 
+  static class MapPropertyParser implements PropertyParser<Map<?, ?>> {
+
+    @Override
+    public Optional<Map<?, ?>> initialize(
+        Properties properties, String separator, Field field, TaskanaProperty taskanaProperty) {
+      if (!Map.class.isAssignableFrom(field.getType())) {
+        throw new SystemException(
+            String.format(
+                "Cannot initialize field '%s' because field type '%s' is not a Map",
+                field, field.getType()));
+      }
+
+      ParameterizedType genericType = (ParameterizedType) field.getGenericType();
+      Type[] actualTypeArguments = genericType.getActualTypeArguments();
+      Class<?> keyClass = (Class<?>) actualTypeArguments[0];
+      Type valueClass = actualTypeArguments[1];
+
+      String propertyKey = taskanaProperty.value();
+      Map<?, ?> mapFromProperties =
+          properties.stringPropertyNames().stream()
+              .filter(it -> it.startsWith(propertyKey))
+              .map(
+                  it -> {
+                    // Keys of the map entry is everything after the propertyKey + "."
+                    String keyAsString = it.substring(propertyKey.length() + 1);
+                    Object key = getStringAsObject(keyAsString, keyClass);
+
+                    // Value of the map entry is the value from the property
+                    String propertyValue = properties.getProperty(it);
+                    Object value = getStringAsObject(propertyValue, separator, valueClass);
+                    return Pair.of(key, value);
+                  })
+              .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+
+      if (mapFromProperties.isEmpty()) {
+        return Optional.empty();
+      } else {
+        return Optional.of(mapFromProperties);
+      }
+    }
+
+    private Object getStringAsObject(String string, String separator, Type type) {
+      if (type instanceof ParameterizedType) {
+        ParameterizedType parameterizedType = (ParameterizedType) type;
+        Type rawType = parameterizedType.getRawType();
+        if (rawType.equals(Set.class)) {
+          return getStringAsSet(
+              string, separator, (Class<?>) parameterizedType.getActualTypeArguments()[0]);
+        }
+      } else if (type instanceof Class) {
+        return getStringAsObject(string, (Class<?>) type);
+      }
+      throw new SystemException("Cannot convert " + string + " to " + type.getTypeName());
```

Comment:
Please use this as an example:
```
        throw new SystemException(
            String.format(
                "Cannot initialize field '%s' because field type '%s' is not a List",
                field, field.getType()));
```

Reply:
done